### PR TITLE
Queue user post endpoint

### LINKED ISSUE #430: Create QueueUser `POST` endpoint
### Pre-requisites:
- [x] I looked through both [open and closed issues](../issues?utf8=✓&q=is%3Aissue) and did not find another request for the same or similar feature.

## Description

We need a new API endpoint which allows authenticated users to join help queues

- Create post request `/api/queue/join` that will create a new `QueueUser`
- It will take a single body key `type` which will be one of the values of the `QueueType` enum
- The error message for an invalid `type` should be generic like `Invalid queue type` so people can't figure out what queues we have enabled

### PR DESCRIPTION
**Pull request by Ulfingrad**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

## Description of Changes

Adds the endpoint responsible for adding users to a queue.

## Related Issues

Closes #430

### CODE REVIEW

File: packages/api/src/api/queue.ts [hunk 1]
```diff
@@ -27,3 +29,39 @@ queue.get('/:type', async (req, res) => {
     res.status(500).send(errorMsg);
   }
 });
+
+queue.post('/join', async (req, res) => {
```
Comment (SpencerKaiser) [inline packages/api/src/api/queue.ts:33]: I'm kinda thinking this makes more sense without the `join`... thoughts?

File: packages/api/src/api/queue.ts [hunk 2]
```diff
@@ -27,3 +29,39 @@ queue.get('/:type', async (req, res) => {
...
+  if (!(type in QueueType)) {
+    res.status(400).send('The queue type entered is invalid');
+    return;
+  }
+
+  try {
+    const currentQueueItems = await req.entityManager.find(QueueUser, {
+      user,
+      status: { $in: ['Pending', 'InProgress'] },
+    });
```
Comment (SpencerKaiser) [inline packages/api/src/api/queue.ts:46]: Let's change this call to a count, that way call is faster
```suggestion
    const userAlreadyInQueue = await req.entityManager.count(QueueUser, {
      user,
      status: { $in: ['Pending', 'InProgress'] },
    }).then((count) => !!count);
```

**Review by SpencerKaiser — COMMENTED** [review]
Super tiny change and one question

**Review by dre8597 — APPROVED** [review]
LGTM 🚀

**Pull request merged**